src/D2L.CodeStyle.Analyzers/Diagnostics.cs

```diff
@@ -757,5 +757,15 @@ public static class Diagnostics {
 			defaultSeverity: DiagnosticSeverity.Error,
 			isEnabledByDefault: true
 		);
+
+		public static readonly DiagnosticDescriptor PinnedTypesMustNotMove = new DiagnosticDescriptor(
+			id: "D2L0102",

```

Comment:
This will likely end up in conflict and I'll bump it closer to when we want to land it.